### DISCUSSION TITLE
Support to collect cmd output

### DISCUSSION THREAD

**Discussion by linrunqi08** [alibaba/ilogtail#782] [Ideas]
Ref：
https://docs.fluentbit.io/manual/pipeline/inputs/exec
https://github.com/influxdata/telegraf/tree/master/plugins/inputs/exec
https://github.com/influxdata/telegraf/tree/master/plugins/inputs/execd

**Comment by linrunqi08** [comment]
插件名：metric_exec

<ne-clipboard data="%7B%22type%22%3A%22fragment%22%2C%22name%22%3A%22%23fragment%22%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22ua81a6c5b%22%2C%22name%22%3A%22tableHole%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22wvPPe%22%2C%22name%22%3A%22table%22%2C%22attrs%22%3A%7B%22rowCount%22%3A7%2C%22colCount%22%3A4%2C%22colWidths%22%3A%5B250%2C250%2C250%2C250%5D%2C%22widthMode%22%3A%22contain%22%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u1b761a47%22%2C%22name%22%3A%22tr%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u626ec9cf%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A0%2C%22cellBgColor%22%3A%22%23F4F5F5%22%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22udaa1cf02%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22ue35040b6%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22%E5%8F%82%E6%95%B0%E5%90%8D%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u81a3e1cb%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A1%2C%22cellBgColor%22%3A%22%23F4F5F5%22%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u753b902a%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22u6f36bb58%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22%E7%B1%BB%E5%9E%8B%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u4ccca6b5%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A2%2C%22cellBgColor%22%3A%22%23F4F5F5%22%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22uc830167a%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22u0f09b563%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22%E6%98%AF%E5%90%A6%E5%BF%85%E5%A1%AB%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u2c9b97de%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A3%2C%22cellBgColor%22%3A%22%23F4F5F5%22%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u6d75f7ac%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22u7fc1db80%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22%E5%AD%97%E6%AE%B5%E8%AF%B4%E6%98%8E%22%7D%5D%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u52c3b710%22%2C%22name%22%3A%22tr%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u4070966f%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A0%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22ufb4b68f8%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22uaee88ebb%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22Command%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u4f1b9648%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A1%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u984845c3%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22uec147fcf%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22String%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u68048e10%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A2%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u89f97d88%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22u0923b3db%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22%E6%98%AF%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22uc946a975%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A3%7D%2C%22children%22%3A%5B%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22uea84c503%22%2C%22name%22%3A%22tr%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22ue954941b%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A0%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22ue6727918%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22ue44bdf81%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22Interval%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u08a0e33a%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A1%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u26e4f55b%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22u263ab6f9%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22int%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u9d271e54%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A2%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22uda1eb5fa%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22u132dc4f9%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22%E6%98%AF%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u0cb7078d%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A3%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u2834f175%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22u307cf0f6%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22%E6%97%B6%E9%97%B4%E9%97%B4%E9%9A%94%22%7D%5D%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u07f9e0cc%22%2C%22name%22%3A%22tr%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u114ba14e%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A0%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u5e0d2721%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22u91085f60%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22IntervalTimeUnit%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u29faf0f5%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A1%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u4d1360fe%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22u243c70ba%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22String%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22ueeacf1ed%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A2%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22ua4e2a646%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22u3c8784bc%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22%E5%90%A6%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22ub08699d6%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A3%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u64a465b7%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22uab0b2c7e%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22%E9%BB%98%E8%AE%A4%E5%80%BC%E4%B8%BAs%E3%80%82%E5%8F%96%E5%80%BC%E5%8C%85%E6%8B%ACms%EF%BC%88%E6%AF%AB%E7%A7%92%EF%BC%89%E3%80%81%E5%92%8Cs%22%7D%5D%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22uf863b3e8%22%2C%22name%22%3A%22tr%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u299a5ce8%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A0%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u84d87bb1%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22u38bfa6b1%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22SplitSep%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u93185759%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A1%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22ue7ed7c28%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22ub2073a8a%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22String%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22ufa009851%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A2%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u3e697b47%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22udaa88aa2%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22%E5%90%A6%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u2e1f1e9e%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A3%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u28b757f8%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22u626434dc%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22%E6%97%A5%E5%BF%97%E7%9A%84%E5%88%86%E9%9A%94%E7%AC%A6%EF%BC%8C%E9%BB%98%E8%AE%A4%E6%98%AF%5C%5C0%22%7D%5D%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u3959ca6a%22%2C%22name%22%3A%22tr%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u4e0b314b%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A0%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22uaeef6bb2%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22u3f08d82f%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22SplitKey%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u4e91a8e7%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A1%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u077ebba3%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22u8c50750a%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22String%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22ud0917118%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A2%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u7011a6dd%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22ubbe3b772%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22%E5%90%A6%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u7c400a43%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A3%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u896cf7be%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22u82ff9f11%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22%E5%88%86%E9%9A%94%E5%90%8E%E7%9A%84%E6%97%A5%E5%BF%97%E5%AD%97%E6%AE%B5%E5%90%8D%EF%BC%8C%E9%BB%98%E8%AE%A4%E6%98%AFcontent%22%7D%5D%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22ub7038d4a%22%2C%22name%22%3A%22tr%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22ud96ff3a9%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A0%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22uc6608532%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22u7025628c%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22TimeoutMilliSeconds%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u347cf3a3%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A1%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u5c0a6b98%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22uadc52f31%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22int%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u869ffacf%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A2%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22uae357c15%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22u5150cd26%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22%E5%90%A6%22%7D%5D%7D%5D%7D%2C%7B%22type%22%3A%22element%22%2C%22id%22%3A%22u2f65cd4f%22%2C%22name%22%3A%22td%22%2C%22attrs%22%3A%7B%22col%22%3A3%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22element%22%2C%22id%22%3A%22ua13e6cf1%22%2C%22name%22%3A%22p%22%2C%22attrs%22%3A%7B%7D%2C%22children%22%3A%5B%7B%22type%22%3A%22text%22%2C%22id%22%3A%22uff88f883%22%2C%22name%22%3A%22%23text%22%2C%22attrs%22%3A%7B%7D%2C%22data%22%3A%22%E5%8D%95%E6%AC%A1%E6%89%A7%E8%A1%8C%E7%9A%84%E6%9C%80%E9%95%BF%E8%B6%85%E6%97%B6%E6%97%B6%E9%97%B4%EF%BC%8C%E6%AF%AB%E7%A7%92%EF%BC%8C%E5%A6%82%E6%9E%9C%E8%AF%A5%E5%8F%82%E6%95%B0%E4%B8%8D%E8%AE%BE%E7%BD%AE%EF%BC%8C%E5%88%99%E4%B8%8D%E8%B6%85%E6%97%B6%EF%BC%8C%E7%AD%89%E4%B8%8B%E6%AC%A1%E6%89%A7%E8%A1%8C%E7%9A%84%E6%97%B6%E5%80%99%EF%BC%8C%E5%BC%BA%E5%88%B6%E5%8F%96%E6%B6%88%22%7D%5D%7D%5D%7D%5D%7D%5D%7D%5D%7D%5D%2C%22attrs%22%3A%7B%7D%7D"></ne-clipboard><div class="lake-content" typography="classic">

参数名 | 类型 | 是否必填 | 字段说明
-- | -- | -- | --
ScriptType | String | 是 |  bash, shell, python
User | String | 是 | 运行命令使用的用户名，只支持非Root用户(建议配置最小权限，只给需要关注的目录/文件rwx权限)
ScriptContent | String | 是 | 脚本内容, 支持原始内容和base64加密的内容, 跟ContentEncoding的字段对应
ContentEncoding| String | 否| PlainText: 不编码; Base64: base64编码，默认PlainText
LineSplitSep | String | 否 | 日志的行分隔符，如果不配置就是全部作为一条数据返回
TimeoutMilliSeconds | int | 否 | 单次执行的最长超时时间，毫秒，如果该参数不设置，则不超时，等下次执行的时候，强制取消
RecordExecTime | Boolean | 否 | 默认是false，是否记录脚本执行时间到数据中，保存字段为execTime
IntervalMs | int | 否 | 执行间隔，metric插件通用字段

</div>


 数据获取核心逻辑样例：
```go
cmd := exec.Command(someCommand, someArg)
pipe, _ := cmd.StdoutPipe()
if err := cmd.Start(); err != nil {
    // handle error
}
go func(p io.ReadCloser) {
    reader := bufio.NewReader(pipe)
    line, err := reader.ReadString('\n')
    for err == nil {
        fmt.Println(line)
        line, err = reader.ReadString('\n')
    }
}(pipe)

if err := cmd.Wait(); err != nil {
    // handle error
}
``` 

指定用户执行逻辑样例:
```go
package main

import (
	"log"
	"os"
	"os/exec"
	"os/user"
	"strconv"
	"syscall"
)

func main() {
	cmd := exec.Command("pwd")
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr

	user, err := user.Lookup("test") //查找指定test用户是否存在，获取Uid和Gid
	if err == nil {
		log.Printf("uid=%s,gid=%s", user.Uid, user.Gid)

		uid, _ := strconv.Atoi(user.Uid)
		gid, _ := strconv.Atoi(user.Gid)

		cmd.SysProcAttr = &syscall.SysProcAttr{}
		cmd.SysProcAttr.Credential = &syscall.Credential{Uid: uint32(uid), Gid: uint32(gid)} //设置执行用户为test
	}

	err = cmd.Run()
	if err != nil {
		log.Fatal(err)
	}
}

```

本插件，采集到的标准输出，会按照LineSep字段做切割，切割每行的数据都存储在content字段中，后续可以使用其他插件对content字段进行解析

### **日志数据处理**
比如脚本输出数据如下
```
Apr 12 10:34:14 iZj6cd7hor3zgygjh7qf14Z systemd: Started Session 9124 of user root.
Apr 12 10:34:14 iZj6cd7hor3zgygjh7qf14Z systemd-logind: Removed session 9124.
```
可以使用正则插件进行解析
```
- Type: processor_regex
    SourceKey: content
    Regex: (\S+\s\d+\s\d\d:\d\d:\d\d)\s(\S+)\s(\S+):\s(.*)
    Keys:
      - time
      - host
      - service
      - message
```


### **时序数据处理**
```
SLS时序数据固定字段如下：
__name__
__value__
__time_nano__
__labels__
``` 

比如脚本输出数据如下
```
2023/04/12 12:59:49 cpu: 0.5
2023/04/12 12:59:59 cpu: 0.6
```
可以使用正则插件进行解析
```
- Type: processor_regex
    SourceKey: content
    Regex: (\d+\/\d+\/\d+\s\d+:\d+:\d+)\s
    Keys:
      - originTime
      - __name__
      - __value__
- Type: processor_appender
    Key: "__labels__"
    Value: "|host#$#{{__host__}}|ip#$#{{__ip__}}"
    SortLabels: true
- Type: processor_strptime
    SourceKey: originTime
    EnablePreciseTimestamp: true
    PreciseTimestampKey: __time_nano__
    PreciseTimestampUnit: ns
```

**Comment by linrunqi08** [comment]
如果标准输出很大的话，应该也不会有问题，如果读取的地方没有读完的话，写入的地方会hang住，等待缓冲队列有空间
https://zhuanlan.zhihu.com/p/453298672
https://github.com/Kevin-fqh/learning-k8s-source-code/blob/master/Linux/(03)Unix-Linux%E7%BC%96%E7%A8%8B%E5%AE%9E%E8%B7%B5%E4%B9%8BIO%E9%87%8D%E5%AE%9A%E5%90%91%E5%92%8C%E7%AE%A1%E9%81%93.md
https://github.com/golang/go/issues/16787

**Comment by linrunqi08** [comment]
https://www.cnblogs.com/liuxinyustu/articles/12068870.html

**Comment by linrunqi08** [comment]
是否需要考虑敏感词过滤？比如kill，rm之类的

**Comment by messixukejia** [comment]
采集的数据怎么转成sls的时序？

**Reply by linrunqi08** [reply]
时序数据固定字段如下：
__name__
__value__
__time_nano__
__labels__
其他字段需要用插件处理，__labels__可以参考如下：
{
  "processors":[
    {
      "type":"processor_appender",
      "detail": {
        "Key": "__labels__",
        "Value": "|host#$#{{__host__}}|ip#$#{{__ip__}}",
        "SortLabels": true
      }
    }
  ]
}

**Comment by EvanLjp** [comment]
是不是给出进制指令默认黑名单， 如rm，touch、mkdir 、rebot、kill 等，基础允许指令范围是无资源操作类的指令

**Comment by EvanLjp** [comment]
SplitKey是不是改成SplitKeys字段，默认参数[]string{"content"}, 支持多个参数配置，结果按分割复分割后给到各个字段，时序模式再需要考虑下模型

**Comment by pj1987111** [comment]
exec权限很大，虽然可以在命令的地方过滤rm，touch、mkdir 、reboot、kill 等。但是可以把命令写到脚本里，exec执行脚本。需要在文档里写好免责声明。

**Comment by yyuuttaaoo** [comment]
Interval | int | 是 | 时间间隔
-- | -- | -- | --
Command | String | 是
IntervalTimeUnit | String | 否 | 默认值为s。取值包括ms（毫秒）、和s
SplitSep | String | 否 | 日志的分隔符，默认是\0
SplitKey | String | 否 | 分隔后的日志字段名，默认是content

Command是否能说明清楚，怎么区分command和args。比如，有个可执行脚本叫"do it.sh"，怎么执行这个命令？
Interval是不是和TimeOut一样一种单位就完事儿了，另外也有限制，不可能支持比如1ms这种吧
SplitSep是什么意思？是指切分多行吗？
SplitKey好像我们其他插件都没有，是不是不提供，如果需要的话，感觉所有input插件应该有个统一的配置项。

另外，这个插件要支持设置环境变量吗？我感觉很可能有用。

**Comment by messixukejia** [comment]
Command 需要支持下直接配置脚本内容，需要确认下采集配置最大长度是否支持

**Comment by pj1987111** [comment]
因为可以执行脚本非常灵活，一些业务性较强的自定义指标采集可以在脚本中实现，然后通过这个cmd插件来调用，每次请求返回的可能会有n个指标，需要对每个指标进行处理并转成sls时序。

**Comment by pj1987111** [comment]
感觉IntervalTimeUnit字段没必要，一般调度都是秒为单位，毫秒场景可能没有。TimeoutMilliSeconds单位也可以是秒，改为Timeout就行了。
SplitSep和SplitKey意义不是特别大，执行返回是什么就行了。因为linux命令返回值千奇百怪而且行很多，拆散反而对排查问题有影响。比如top，free等命令。

**Comment by yyuuttaaoo** [comment]
example plugin: https://github.com/alibaba/ilogtail/blob/main/plugins/input/example/metric_example.go
collector API: https://github.com/alibaba/ilogtail/blob/main/pkg/pipeline/collector.go
log v1 model: https://github.com/alibaba/ilogtail/blob/main/pkg/protocol/proto/sls_logs.proto
register: https://github.com/alibaba/ilogtail/blob/main/plugins.yml
dev env: https://github.com/alibaba/ilogtail/blob/main/docs/cn/developer-guide/development-environment.md
debug: https://github.com/alibaba/ilogtail/blob/main/docs/cn/developer-guide/plugin-development/pure-plugin-start.md

**Comment by yyuuttaaoo** [comment]
close as resolved in #925
